app/main.py

```diff
@@ -1,3 +1,98 @@
+from typing import Union
+
+
 class Distance:
-    # Write your code here
-    pass
+    def __init__(self, km: Union[int, float]) -> None:
+        self.km = km
+
+    def __str__(self) -> str:
+        return f"{Distance.__name__}: {self.km} kilometers."
+
+    def __repr__(self) -> str:
+        return f"{Distance.__name__}(km={self.km})"
+
+    def __add__(self, other: Union["Distance", int, float]
+                ) -> Union["Distance", TypeError]:
```

Comment:
```suggestion
    def __add__(self, other: Distance | int | float) -> Distance:
```

Reply:
this  approach using `Distance | int | float` is concise, clear, and follows the modern conventions of type hinting in Python, providing a straightforward and expressive way

Reply:
Fix in all places